apps/test-low-memory.sh

```diff
@@ -30,8 +30,36 @@ function linuxNumProcs() {
   echo $procs
 }
 
+function macNumProcs() {
+  # FIXME: top's PhysMem unused is not a great metric for available memory:
+  # it waaaaay underestimates how much memory is actually available
+  # but no equivalent to Linux's MemAvailable is available on Mac.
+  
+  local min_procs=2
+
+  # extract unused ###G from `top -l1` line like:
+  # PhysMem: 13G used (3248M wired, 4117M compressor), 3G unused
+  local unused_mem_regex='^PhysMem.* ([0-9]+)G unused.*$'
+  if [[ $(top -l1 | grep -e '^PhysMem') =~ $unused_mem_regex ]]; then
+    local unusedMemGB=${BASH_REMATCH[1]}
+    local mem_procs=$(( unusedMemGB * 1024 / ${MEM_PER_PROCESS}))
+    # Overshoot by one process to get better performance. Works because 
+    # using top's PhysMem undershoots by a LOT on lower-mem macs.
+    mem_procs=$(( mem_procs + 1 ))
+  else
+    echo "Couldn't parse `top -l1` output to find amount of unused memory."
+    local mem_procs=$min_procs
+  fi
+
+  nprocs=$(nproc)
+  mem_procs=$(( mem_procs < min_procs ? min_procs : mem_procs ))
```

Comment:
would we want `mem_procs` to be _raised_ to `min_procs` in the event that the logic above set it to `1`?

Reply:
Hah! I was literally sitting here deadlocked pondering this very point when you commented 😆 Where I landed is that we probably DO want to do at least 2 parallel instances even in the (very hypothetical since there are no single core macs and we don't currently using docker+darwin) instance of only having one CPU.

Our tests are not CPU bound when running (they'll be IO bound by far), and max number of instances will be even more determined by available RAM in most cases.